community/pike/APKBUILD

```diff
@@ -0,0 +1,200 @@
+}
+
+package() {
+	cd "$builddir"
+	for pkg in $(printf "%s" "$subpackages" | tr ' ' '\n'); do
+		depends="$depends $pkg=$pkgver-r$pkgrel"
+	done
+	make INSTALLARGS="--traditional" buildroot="$pkgdir" install
+}
+
+_move_item() {
+	local bn=$(basename "$1")
+	local dn=$(dirname "$1")
+	local dst="$2/$dn"
+	mkdir -p "$dst"
+	mv "$1" "$dst/$bn"
+}
+
+_install() {
+	local bn=$(basename "$1")
+	local dn=$(dirname "$1")
+	find "$dn" -maxdepth 1 -name "$bn" | \
+		while IFS= read item; do _move_item "$item" "$2"; done
+}
+
+bzip2() {
+	cd "$pkgdir"
+	pkgdesc="$pkgname bzip2 support"
+	depends="pike-core=$pkgver-r$pkgrel bzip2"
+	_install "usr/lib/pike/modules/*Bz2.*" "$subpkgdir"
+}
+
+gdbm() {
+	cd "$pkgdir"
+	pkgdesc="$pkgname GDBM support"
+	depends="pike-core=$pkgver-r$pkgrel gdbm"
+	_install "usr/lib/pike/modules/*Gdbm.*" "$subpkgdir"
+}
+
+gl() {
+	cd "$pkgdir"
+	pkgdesc="$pkgname OpenGL/GLUT support"
+	depends="pike-core=$pkgver-r$pkgrel freeglut mesa-gl libsm"
+	_install "usr/lib/pike/modules/*GL*" "$subpkgdir"
+}
+
+gtk() {
+	cd "$pkgdir"
+	pkgdesc="$pkgname GTK2 support"
+	depends="pike-image=$pkgver-r$pkgrel libglade gtk+2.0"
+	_install "usr/lib/pike/modules/*GTK2*" "$subpkgdir"
+	_install "usr/lib/pike/modules/Tools.pmod/PV.pike" "$subpkgdir"
+}
+
+image() {
+	cd "$pkgdir"
+	pkgdesc="$pkgname image handling modules"
+	depends="pike-core=$pkgver-r$pkgrel giflib-utils libjpeg tiff libwebp"
+	_install "usr/lib/pike/?.?/modules/Image.pmod" "$subpkgdir"
+	_install "usr/lib/pike/modules/Image.so" "$subpkgdir"
+	_install "usr/lib/pike/modules/*Image_[!S]*.so" "$subpkgdir"
+	_install "usr/lib/pike/modules/*Image*.pmod" "$subpkgdir"
+	_install "usr/lib/pike/modules/Graphics.pmod" "$subpkgdir"
+	_install "usr/lib/pike/modules/Colors.pmod" "$subpkgdir"
+	_install "usr/lib/pike/modules/Protocols.pmod/X.pmod/XImage.pmod" "$subpkgdir"
+}
+
+pcre() {
+	cd "$pkgdir"
+	pkgdesc="$pkgname PCRE support"
+	depends="pike-core=$pkgver-r$pkgrel pcre"
+	_install "usr/lib/pike/modules/*Regexp_PCRE.*" "$subpkgdir"
+}
+
+sdl() {
+	cd "$pkgdir"
+	pkgdesc="$pkgname SDL support"
+	depends="pike-image=$pkgver-r$pkgrel sdl sdl_mixer"
+	_install "usr/lib/pike/modules/*SDL.*" "$subpkgdir"
+}
+
+svg() {
+	cd "$pkgdir"
+	pkgdesc="$pkgname SVG support"
+	depends="pike-core=$pkgver-r$pkgrel librsvg"
+	_install "usr/lib/pike/modules/_Image_SVG.so" "$subpkgdir"
+}
+
+sqlite() {
+	cd "$pkgdir"
+	pkgdesc="$pkgname SQLite support"
+	depends="pike-core=$pkgver-r$pkgrel sqlite-libs"
+	_install "usr/lib/pike/modules/*SQLite.*" "$subpkgdir"
+	_install "usr/lib/pike/modules/Sql.pmod/sqlite*" "$subpkgdir"
+}
+
+doc() {
+	cd "$pkgdir"
+	mkdir -p usr/share/doc/
+	mv "$builddir/refdoc/traditional_manual/" usr/share/doc/pike/
+	rm -rf usr/doc/
+	default_doc
+}
+
+core() {
+	cd "$pkgdir"
+	pkgdesc="Pike's core files and executables"
+	depends="gmp nettle nettle-utils zlib readline libgcrypt"
+	mkdir -p "$subpkgdir"
+	mv usr "$subpkgdir/"
```

Comment:
I'm going to suggest leaving `usr/bin` in `pike` package.

```
mkdir -p "$subpkgdir/usr"
mv usr/include usr/lib "$subpkgdir/usr/"
```

Reply:
Ok, so maybe get rid of `pike-core` and leave core stuff in `pike`, and add something like `pike-full` meta package that will pull in all additional `pike-` packages?

Reply:
I'd suggest `pike-modules` so that you could choose to install `pike` and specific individual modules or `pike` and `pike-modules` which would depend on all the split up modules.